### DISCUSSION TITLE
AWS lambda function not identify the handler in FastAPI project

### DISCUSSION THREAD

**Discussion by parth-patel97** [Kludex/mangum#290] [Q&A]
When I try to upload the FastAPI on the lambda function I get this error. Would anyone be able to help me with this?

My Porject structure :
README.md
main.py
requirements.txt
routes/api.py
src/dependencies/config.py
src/endpoints/textGenerator.py

My main.py :- 
```
from fastapi import FastAPI,Depends
from routes.api import router as api_router
from mangum import Mangum

app = FastAPI()
app.include_router(api_router)

handler = Mangum(app=app)

```

{
  "errorMessage": "The adapter was unable to infer a handler to use for the event. This is likely related to how the Lambda function was invoked. (Are you testing locally? Make sure the request payload is valid for a supported handler.)",
  "errorType": "RuntimeError",
  "stackTrace": [
    "  File \"/opt/python/mangum/adapter.py\", line 76, in __call__\n    handler = self.infer(event, context)\n",
    "  File \"/opt/python/mangum/adapter.py\", line 68, in infer\n    raise RuntimeError(  # pragma: no cover\n"
  ]
}

**Comment by renkasiyas** [comment]
Any on this?

**Comment by ghost** [comment]
Hey @parth-patel97

Faced the same issue a couple of days ago and this [PR](https://github.com/jordaneremieff/mangum/pull/293) helped me a lot, it seems to be a bug while trying to handle a proxy lambda, nothing about this is documented tho.

Hope that helps!
